### PR TITLE
Websocket telnet fixes

### PR DESCRIPTION
**Pull request by icewolfz**
This removed the extra comment and makes the changes to use maybe_schedule_user_command instead of how the telnet block does it

### CODE REVIEW

**Comment by thefallentree** [comment]
```
*** CID 1563407:  API usage errors  (PW.TOO_MANY_PRINTF_ARGS)
/home/runner/work/fluffos/fluffos/src/packages/core/telnet_ext.cc: 176 in ()
170     void f_send_msdp_variable() {
171       auto *ip = current_object->interactive;
172       if (ip && ip->telnet) {
173         switch(sp->type) {
174           case T_STRING:
175             telnet_begin_sb(ip->telnet, TELNET_TELOPT_MSDP);
>>>     CID 1563407:  API usage errors  (PW.TOO_MANY_PRINTF_ARGS)
>>>     the format string ends before this argument
176             telnet_printf(ip->telnet, "\x01%s\x02", (sp - 1)->u.string, sp->u.string);
177             telnet_send(ip->telnet, sp->u.string, SVALUE_STRLEN(sp));
178             telnet_finish_sb((ip->telnet));
179             break;
180           case T_NUMBER:
181             telnet_begin_sb(ip->telnet, TELNET_TELOPT_MSDP);


```

got this from the static lint , it does seems 176 is missing another %s ?

**Comment by thefallentree** [comment]
actually, it appears that 176 should have sp->u.string deleted

**Comment by thefallentree** [comment]
also,  please write something similar to GMCP in docs/apply

**Comment by thefallentree** [comment]
reference: https://www.fluffos.info/apply/interactive/gmcp.html#gmcp

**Comment by icewolfz** [comment]
i will push a fix i tested it must have hit undo or pushed old file

**Comment by icewolfz** [comment]
yeah i see what i did it was a bad file i changed it on my vm testing but forget to pull the fix it should not have the 2nd as i moved that to separate line to avoid printf string limits

**Comment by icewolfz** [comment]
ok msdp should be fixed now, i am building a copy to be 100% sure but this push should fix it

**Comment by icewolfz** [comment]
I added the docs for all the msdp efuns, and i added the doc for has_msp as well

**Comment by icewolfz** [comment]
Okay i think is everything you wanted, added all the docs, fixed the send_msdp_variable, adds the  maybe_schedule_user_command, and removes the extra comment, i tested the fixes on my local copy and i could send msdp with no issues

**Comment by icewolfz** [comment]
I updated the testsuite to test MSDP, figured since adding all the docs you would want the testsuite to have the support as well

File: docs/efun/index.md [hunk 1]
```diff
@@ -199,11 +199,14 @@ title: EFUN
 * [has_gmcp](interactive/has_gmcp.html)
 * [has_mxp](interactive/has_mxp.html)
 * [has_zmp](interactive/has_zmp.html)
+* [has_msp](interactive/has_msp.html)
+* [has_msdp](interactive/has_masdp.html)
```
Comment (michaelprograms) [inline docs/efun/index.md:203]: Shouldn't this be `has_msdp.html`?

Comment (icewolfz) [reply docs/efun/index.md:203]: yeah it should i miss it, i probably hit the a/s key at the same time and didnt think to double check the link assuming it was right, fixing it now

File: docs/apply/interactive/msdp_enable.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
+layout: doc
+title: interactive / msdp_enable
+---
+# gmcp_enable
```
Comment (thefallentree) [inline docs/apply/interactive/msdp_enable.md:5]: copy/paste error :-D

File: docs/efun/interactive/has_msp.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+---
+layout: doc
+title: interactive / has_msp
+---
+# has_mxp
```
Comment (thefallentree) [inline docs/efun/interactive/has_msp.md:5]: copy/paste error :-D

**Review by thefallentree — APPROVED** [review]

File: docs/efun/index.md (hunk 1 above)
Comment (thefallentree) [reply docs/efun/index.md]: the index are actually auto generated, just ran `./update_index.sh`

File: docs/efun/interactive/has_msp.md (hunk 3 above)
Comment (icewolfz) [reply docs/efun/interactive/has_msp.md]: Fixed the 3 copy pate errors

**Pull request merged**